app/controllers/ingredients_controller.rb

```diff
@@ -0,0 +1,71 @@
+class IngredientsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_ingredient, only: %i[show edit update destroy]
+
+  def index
+    @ingredients = current_user.ingredients.all
+  end
+
+  def show; end
+
+  def new
+    @ingredient = current_user.ingredients.build
+    @recipe = current_user.recipes.find(params[:recipe_id])
+    @foods = Food.all
+  end
+
+  def create
+    @ingredient = current_user.ingredients.build(ingredient_params)
+    @recipe = current_user.recipes.find(params[:recipe_id])
+    respond_to do |format|
+      if @ingredient.save
+        format.html { redirect_to recipe_path(@recipe), notice: 'Ingredient saved successfully' }
+        format.json { render :show, status: :created, location: @ingredient }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @ingredient.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def edit
+    @ingredient = current_user.ingredients.find(params[:id])
+    @recipe = current_user.recipe
+  end
+
+  def update
```

Comment:
- [ ] We are missing the update method on the front end as per the wireframe, the user needs to be able to modify the ingredient, see? please fix this
- [ ] YOU
- [ ] 
<img width="480" alt="2" src="https://github.com/AbbasSarwar/Recipe-app/assets/77843050/31d116ce-e3dc-4565-aa09-5029d841a103">

- [ ] EXPECTED

<img width="655" alt="1" src="https://github.com/AbbasSarwar/Recipe-app/assets/77843050/5f859516-06c1-4a7f-ac40-fdba22f75dd7">

Reply:
Done as advised.